scripts/reference-tables.js

```diff
@@ -0,0 +1,409 @@
+  fs.readdirSync(path).forEach(function (file) {
+    var newPath = path + '/' + file;
+
+    var stats = fs.statSync(newPath);
+
+    if (stats.isDirectory()) {
+      findHTMLFiles(newPath);
+    }
+
+    if (stats.isFile() &&
+        (newPath.indexOf('.html') > 0) &&
+        (newPath.indexOf('index.html') < 0) &&
+        (newPath.indexOf('landmark') < 0)) {
+      count += 1;
+
+      var data = fs.readFileSync(newPath, 'utf8');
+
+      var ref = newPath;
+      var title = getTitle(data);
+      var roles = getRoles(data);
+      var props = getPropertiesAndStates(data);
+
+      console.log('\nFile ' + count + ': ' + ref);
+      console.log('Title  ' + count + ': ' + title);
+      console.log('Roles  ' + count + ': ' + roles);
+      console.log('Props  ' + count + ': ' + props);
+
+      var example = {};
+
+      example.title = title;
+      example.ref = ref;
+
+      addExampleToRoles(roles, example);
+
+      addExampleToPropertiesAndStates(props, example);
+    }
+  });
+};
+
+findHTMLFiles(examplesDirectory);
+
+var res = fs.readdir('.');
+
+for (r in res) {
+
+  // Handle landmark examples separately
+  if (res[r].indexOf('landmark') < 0) {
+  }
+}
+
+// Add landmark examples, since they are a different format
+
+addLandmarkRole('banner', false, 'Banner Landmark', examplesDirectory + 'landmarks/banner.html');
+addLandmarkRole('complementary', true, 'Complementary Landmark', examplesDirectory + 'landmarks/complementary.html');
+addLandmarkRole('contentinfo', false, 'Contentinfo Landmark', examplesDirectory + 'landmarks/contentinfo.html');
+addLandmarkRole('form', true, 'Form Landmark', examplesDirectory + 'landmarks/form.html');
+addLandmarkRole('main', true, 'Main Landmark', examplesDirectory + 'landmarks/main.html');
+addLandmarkRole('navigation', true, 'Navigation Landmark', examplesDirectory + 'landmarks/navigation.html');
+addLandmarkRole('region', true, 'Region Landmark', examplesDirectory + 'landmarks/region.html');
+addLandmarkRole('search', true, 'Search Landmark', examplesDirectory + 'landmarks/search.html');
+
+var exampleIndexFile = fs.readFileSync(fileNameTemplate, function (err) {
+  console.log('Error reading aria index:', err);
+});
+
+var sorted = [];
+
+for (role in indexOfRoles) {
+  sorted.push(role);
+}
+
+sorted.sort();
+
+var html = sorted.reduce(function (set,role) {
+  var examples = indexOfRoles[role];
+
+  var examplesHTML = '';
+  if (examples.length === 1) {
+    examplesHTML = '    <a href="' + examples[0].ref + '">' + examples[0].title + '</a>';
+  }
+  else {
+    function exampleListItem (item) { return '      <li><a href="' + item.ref + '">' + item.title + '</a></li>';};
+    examplesHTML = '    <ul>' + examples.map(exampleListItem).join('\n') + '</ul>';
+  }
+  return set + '<tr>\n  <td><code>' + role + '</code></td>\n  <td>' + examplesHTML + '</td>\n</tr>';
+}, '');
+
+exampleIndexFile = replaceSection('examples_by_role_tbody', exampleIndexFile, html);
+
+sorted = [];
```

Comment:
The linter we use flas let and const as errors

Reply:
OK, there should probably be some overrides for the scripts in this folder since they're node rather than browser based. That can be done afterwards though